packages/tools/src/tools/annotation/planarFreehandROITool/openContourEditLoop.ts

```diff
@@ -0,0 +1,548 @@
+import { getEnabledElement } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { state } from '../../../store';
+import { Events } from '../../../enums';
+import {
+  resetElementCursor,
+  hideElementCursor,
+} from '../../../cursors/elementCursor';
+import type { EventTypes, Annotation } from '../../../types';
+import { vec3, vec2 } from 'gl-matrix';
+import { polyline } from '../../../utilities/math';
+import triggerAnnotationRenderForViewportIds from '../../../utilities/triggerAnnotationRenderForViewportIds';
+
+const { addCanvasPointsToArray, getSpacingAndXYDirections } = polyline;
+
+/**
+ * Activates the open contour edit event loop.
+ */
+function activateOpenContourEdit(
+  evt: EventTypes.MouseDownActivateEventType,
+  annotation: Annotation,
+  viewportIdsToRender: string[]
+): void {
+  this.isEditingOpen = true;
+
+  const eventDetail = evt.detail;
+  const { currentPoints, element } = eventDetail;
+  const canvasPos = currentPoints.canvas;
+  const enabledElement = getEnabledElement(element);
+  const { viewport } = enabledElement;
+
+  const prevCanvasPoints = annotation.data.polyline.map(viewport.worldToCanvas);
+
+  const { spacing, xDir, yDir } = getSpacingAndXYDirections(
+    viewport,
+    this.configuration.subPixelResolution
+  );
+
+  this.commonEditData = {
```

Comment:
we use this.editData elsewhere in the annotation tools, is this `commonEditData` different than `editData`?

Reply:
This is because I had commonEdit data and some edit data specific to the type of edit. I've actually changed this, so it can be edit data now. Will change.